src/Components/Browser.JS/src/Platform/Circuits/CircuitManager.ts

```diff
@@ -0,0 +1,129 @@
+import { internalFunctions as uriHelperFunctions } from '../../Services/UriHelper';
+import { ComponentDescriptor, MarkupRegistrationTags, StartComponentComment, EndComponentComment } from './ComponentDescriptor';
+
+export class CircuitDescriptor {
+  public circuitId: string;
+  public components: ComponentDescriptor[];
+
+  public constructor(circuitId: string, components: ComponentDescriptor[]) {
+    this.circuitId = circuitId;
+    this.components = components;
+  }
+
+  public reconnect(reconnection: signalR.HubConnection): Promise<boolean> {
+    return reconnection.invoke<boolean>('ConnectCircuit', this.circuitId);
+  }
+}
+
+
+export default class CircuitManager {
```

Comment:
I know this is nitpicky, but why have a class for this and not just a single `export function`?

Reply:
done